plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -222,28 +223,29 @@ func (c *MaxPDVolumeCountChecker) filterVolumes(volumes []api.Volume, namespace
 	return nil
 }
 
-func (c *MaxPDVolumeCountChecker) predicate(pod *api.Pod, meta interface{}, nodeInfo *schedulercache.NodeInfo) (bool, error) {
+func (c *MaxPDVolumeCountChecker) predicate(pod *api.Pod, meta interface{}, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
+	predicateFails := make([]algorithm.PredicateFailureReason, 0)
```

Comment:
Please avoid allocating this slice when possible - in most of cases it will be unnecessary allocation that will; be slowing us down.